library/core/src/hint.rs

```diff
@@ -51,18 +51,54 @@ pub const unsafe fn unreachable_unchecked() -> ! {
     unsafe { intrinsics::unreachable() }
 }
 
-/// Emits a machine instruction hinting to the processor that it is running in busy-wait
-/// spin-loop ("spin lock").
+/// Emits a machine instruction to signal the processor that it is running in
+/// a busy-wait spin-loop ("spin lock").
+///
+/// Upon receiving the spin-loop signal the processor can optimize its behavior by,
+/// for example, saving power or switching hyper-threads.
+///
+/// This function is different from [`thread::yield_now`] which directly
+/// yields to the system's scheduler, whereas `spin_loop` does not interact
+/// with the operating system.
+///
+/// A common use case for `spin_loop` is implementing bounded optimistic
+/// spinning in a CAS loop in synchronization primitives. To avoid problems
+/// like priority inversion, it is strongly recommended that the spin loop is
+/// terminated after a finite amount of iterations and an appropriate blocking
+/// syscall is made.
+///
+/// **Note**: On platforms that do not support receiving spin-loop hints this
+/// function does not do anything at all.
+///
+/// # Examples
 ///
-/// For a discussion of different locking strategies and their trade-offs, see
-/// [`core::sync::atomic::spin_loop_hint`].
+/// ```
+/// use std::sync::atomic::{AtomicBool, Ordering};
+/// use std::sync::Arc;
+/// use std::{hint, thread};
+///
+/// // Baby not born yet
+/// let live = Arc::new(AtomicBool::new(false));
+///
+/// // Mum is in the operation room
+/// let live_clone = live.clone();
+/// let _thread = thread::spawn(move || {
+///     // Long process, waiting for baby to release
+///     live_clone.store(true, Ordering::Release);
+/// });
 ///
-/// **Note**: On platforms that do not support receiving spin-loop hints this function does not
-/// do anything at all.
+/// // Dad is busy waiting for news
+/// while live.load(Ordering::Acquire) {
+///     // Dad is busy waiting in a loop
+///     hint::spin_loop();
+/// }
+///
+/// // Baby is out
+/// ```
 ///
-/// [`core::sync::atomic::spin_loop_hint`]: crate::sync::atomic::spin_loop_hint
+/// [`thread::yield_now`]: ../../std/thread/fn.yield_now.html
 #[inline]
-#[unstable(feature = "renamed_spin_loop", issue = "55002")]
+#[stable(feature = "renamed_spin_loop", since = "1.47.0")]
```

Comment:
```suggestion
#[stable(feature = "renamed_spin_loop", since = "1.48.0")]
```

Reply:
1.48 has already promoted to beta, this should be 1.49.
```suggestion
#[stable(feature = "renamed_spin_loop", since = "1.49.0")]
```